docs/DevGuide/BuildSystem.md

```diff
@@ -69,15 +69,33 @@ add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
 target_link_libraries(
   ${LIBRARY}
-  <list_of_interfaces>
+  PUBLIC
+  <list_of_public_libraries>
+  PRIVATE
+  <list_of_private_libraries>
+  INTERFACE
+  <list_of_interface_libraries>
   )
 ```
-where `<list_of_interfaces>` is an alphabetized list of lines of the form
+where each `<list_of_X_libraries>` is an alphabetized list
+of libraries of the form
 ```
-  INTERFACE SOME_LIBRARY
+  SOME_LIBRARY
+  SOME_OTHER_LIBRARY
+  YET_ANOTHER_LIBRARY
```

Comment:
Library names are CamelCase.  It's confusing enough having syntax and "variables" to be replaced both in all-caps in this section, and I think we can use the standard capitalization here and still be clear.